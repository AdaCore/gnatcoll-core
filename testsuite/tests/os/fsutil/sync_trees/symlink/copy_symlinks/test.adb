with Test_Assert;
with Ada.Text_IO;
with GNATCOLL.OS.FSUtil; use GNATCOLL.OS.FSUtil;
with Tree_Helper;

function Test return Integer is
   package A renames Test_Assert;
   package IO renames Ada.Text_IO;

   Src : constant UTF8.UTF_8_String := "src_top_dir";
   Dst : constant UTF8.UTF_8_String := "dst_top_dir";
begin
   IO.Put_Line ("Test for GNATCOLL.OS.FSUtil.Sync_Trees");

   A.Assert (
     Sync_Trees
       (Src, Dst, Mode => TIMESTAMP, Symlink_Mode => COPY_SYMLINKS));

   Tree_Helper.Display_Trees (Src, Dst);
   Tree_Helper.Compare_Tree_Files_Content (Src, Dst);

   return A.Report;
end Test;
