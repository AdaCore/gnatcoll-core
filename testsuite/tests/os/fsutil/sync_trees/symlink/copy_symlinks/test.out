Test for GNATCOLL.OS.FSUtil.Sync_Trees
OK   test.adb:15
Source tree
===========
DIR src_top_dir/dir1
DIR src_top_dir/dir2
DIR src_top_dir/dir3
FILE src_top_dir/dir1/file1
FILE src_top_dir/dir3/file3
FILE src_top_dir/file1
SYMLINK src_top_dir/abs_link_to_missing_file -> /tmp/missing-file
SYMLINK src_top_dir/common_abs_link_to_missing_file -> /tmp/missing-file
SYMLINK src_top_dir/common_link_to_missing_file -> ./missing-file
SYMLINK src_top_dir/dir1/dir3_link -> ../dir3
SYMLINK src_top_dir/file3_link -> ./dir3/file3
SYMLINK src_top_dir/link_to_missing_file -> ./missing-file
SYMLINK src_top_dir/relative_external_dir_link -> ../src_top_dir/dir1
SYMLINK src_top_dir/symlink_in_src_and_dir_in_dst -> /tmp/missing-file

Destination tree
================
DIR dst_top_dir/dir1
DIR dst_top_dir/dir2
DIR dst_top_dir/dir3
FILE dst_top_dir/dir1/file1
FILE dst_top_dir/dir3/file3
FILE dst_top_dir/file1
SYMLINK dst_top_dir/abs_link_to_missing_file -> /tmp/missing-file
SYMLINK dst_top_dir/common_abs_link_to_missing_file -> /tmp/missing-file
SYMLINK dst_top_dir/common_link_to_missing_file -> ./missing-file
SYMLINK dst_top_dir/dir1/dir3_link -> ../dir3
SYMLINK dst_top_dir/file3_link -> ./dir3/file3
SYMLINK dst_top_dir/link_to_missing_file -> ./missing-file
SYMLINK dst_top_dir/relative_external_dir_link -> ../src_top_dir/dir1
SYMLINK dst_top_dir/symlink_in_src_and_dir_in_dst -> /tmp/missing-file

Content checking
================
<=== TEST PASSED ===>
